classes/Mail.php

```diff
@@ -408,12 +413,13 @@ function ($carry, $item) {
                     PrestaShopLogger::addLog(
                         Context::getContext()->getTranslator()->trans(
                             'Error - The following e-mail template is missing: %s',
-                            [$templatePath.$isoTemplate.'.html'],
+                            [$isoTemplateBasePath.$isoTemplate.'.html'],
                             'Admin.Advparameters.Notification'
                         )
                     );
                 } else {
                     $templatePathExists = true;
+                    $templatePath = $isoTemplateBasePath;
```

Comment:
Hum, i will override on each loop and may the line 388 useless :/

Reply:
You passed `$templatePath` as `Mail::send` parameters, it should be used on line `388` as you did, but with your modification, only the first `iso` used the parameter you set.

Reply:
Ok, 
according to what I see some considerations:

isoArray is built with at least 1 isoCode (“en”), up to three isoCodes: 

-	isoCode for the passed lang
-	default isoCode
“en”

Let’s examine this last case.
Foreach loop determines if for any element of the array the correspondent template exists.
If at least one template exists (according to HTML/TXT email template configuration) the method breaks loop and continues its execution,
with the current iso template.

I moved the templatePath validity check outside, thus I don’t have to make an assignation for each loop cycle.
 
Template file presence check is performed always in the same path (passed template path or, if null/empty, default template path).
As it is written we have no fallbacks: if template path is passed, but no files exist in that path, the method rises an error instead of checking
in the default template path and proceeding.

I made a new push, adjusting the code.